### PR TITLE
Adds version check calls on client

### PR DESCRIPTION
**Pull request by Franreno**
This PR handles the version check between the client and the API. It follows the [Semantic Versioning](https://semver.org) convention. (MAJOR.MINOR.PATCH) 
The check is done when the package is imported and:
- Exits if the `major` version is mismatched with the API.
- Warns the user if there is a new version for the `minor` and `patch` differences. 

The API and client follows versions should be updated with a new tag creation or a new release.

### CODE REVIEW

**Comment by lneves-ai** [comment]
the concern raised by @hpenedones that there is an api call when importing the library is still not resolved, right? can it be done lazyly?

**Comment by Franreno** [comment]
> the concern raised by @hpenedones that there is an api call when importing the library is still not resolved, right? can it be done lazyly?

Yes, it still not complete yet. I'm fininshing this part today. 
I have talked to @luispcunha and we think that the best way to approach this it wrap the `validate_api_key` function with some kind of singleton, so that it can check, only once, the version of the client and backend.

File: inductiva/api/methods.py [hunk 1]
```diff
@@ -33,6 +34,11 @@ def validate_api_key(api_key: Optional[str]) -> Configuration:
             "Set it in the code with \"inductiva.api_key = <YOUR_SECRET_KEY>\""
             " or set the INDUCTIVA_API_KEY environment variable.")
 
+    # Perform version check only on first invocation
+    if not hasattr(validate_api_key, "version_checked"):
```
[the remark this answers is not in the document]
Comment (Franreno) [reply inductiva/api/methods.py:38]: The variable `version_checked` is being attributed to the validate_api_key function object only when it is called on the first time (lazy calling). The variable shouldn't exist before the first call to the function, this avoids using global variables and keeps the state encapsulated within the function.

File: inductiva/utils/version_check.py [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+"""
+
+
+class VersionCheckException(Exception):
+
+    def __init__(self, message: str):
+        super().__init__(message)
+
+
+def compare_versions(inductiva_version: str, backend_version: str):
```
[the remark this answers is not in the document]
Comment (Franreno) [reply inductiva/utils/version_check.py:14]: Yes it did. Im going to remove the file.

File: inductiva/tests/test_compare_version.py [hunk 3]
```diff
@@ -21,13 +20,13 @@ def mock_api_client():
...
 # Test successful version match
-def test_version_match(mock_api_client):
+def test_version_match(mock_api_client):  # pylint: disable=unused-argument,redefined-outer-name
     # No exception should be raised
     methods.compare_client_and_backend_versions("1.0.0")
 
 
 # Test version mismatch
-def test_version_mismatch(mock_api_client):
+def test_version_mismatch(mock_api_client):  # pylint: disable=redefined-outer-name
```
[the remark this answers is not in the document]
Comment (Franreno) [reply inductiva/tests/test_compare_version.py]: Thanks for the recomendation, I have fixed these issues.

**Review by lneves-ai — APPROVED** [review]

**Pull request merged**